diff --git a/Makefile b/Makefile
index 0cb28be..5927ab3 100644
--- a/Makefile
+++ b/Makefile
@@ -40,19 +40,28 @@ EXEPATH = ./
 .CFILES   = $(MAINFILES) $(MUSASHIFILES) $(MUSASHIGENCFILES)
 .OFILES   = $(.CFILES:%.c=%.o) a314/a314.o
 
-CC        = gcc
-CXX       = g++
-WARNINGS  = -Wall -Wextra -pedantic
+CC         ?= gcc
+CXX        ?= g++
+PKGCONFIG  ?= pkg-config
+
+CFLAGS     ?= -march=armv8-a -mfloat-abi=hard -mfpu=neon-fp-armv8 -O3
+LDFLAGS    ?= -L/usr/local/lib
+MIN_CFLAGS := $(CFLAGS)
+
+WARNINGS    = -Wall -Wextra -pedantic
+
+CFLAGS     := $(CFLAGS) $(WARNINGS) -I. $(shell $(PKGCONFIG) --cflags bcm_host) -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE
+LDFLAGS    := $(LDFLAGS) $(WARNINGS) $(shell $(PKGCONFIG) --cflags bcm_host) -lraylib -lasound
 
 ifeq ($(PLATFORM),PI3_BULLSEYE)
-	LFLAGS    = $(WARNINGS) -L/usr/local/lib -L/opt/vc/lib -L./raylib_drm -lraylib -lGLESv2 -lEGL -lgbm -ldrm -ldl -lstdc++ -lvcos -lvchiq_arm -lvchostif -lasound
-	CFLAGS    = $(WARNINGS) -I. -I./raylib -I/opt/vc/include/ -march=armv8-a -mfloat-abi=hard -mfpu=neon-fp-armv8 -O3 -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -lstdc++ $(ACFLAGS)
+	LDFLAGS += -L./raylib_drm -lGLESv2 -lEGL -lgbm -ldrm -ldl -lvchostif
+	CFLAGS  += -I./raylib
 else ifeq ($(PLATFORM),PI4)
-	LFLAGS    = $(WARNINGS) -L/usr/local/lib -L/opt/vc/lib -L./raylib_pi4_test -lraylib -lGLESv2 -lEGL -lgbm -ldrm -ldl -lstdc++ -lvcos -lvchiq_arm -lvchostif -lasound
-	CFLAGS    = $(WARNINGS) -DRPI4_TEST -I. -I./raylib_pi4_test -I/opt/vc/include/ -march=armv8-a -mfloat-abi=hard -mfpu=neon-fp-armv8 -O3 -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -lstdc++ $(ACFLAGS)
+	LDFLAGS += -L./raylib_pi4_test -lGLESv2 -lEGL -lgbm -ldrm -ldl -lvchostif
+	CFLAGS  += -I./raylib_pi4_test -DRPI4_TEST
 else
-	CFLAGS    = $(WARNINGS) -I. -I./raylib -I/opt/vc/include/ -march=armv8-a -mfloat-abi=hard -mfpu=neon-fp-armv8 -O3 -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -lstdc++ $(ACFLAGS)
-	LFLAGS    = $(WARNINGS) -L/opt/vc/lib -L./raylib -lraylib -lbrcmGLESv2 -lbrcmEGL -lbcm_host -lstdc++ -lvcos -lvchiq_arm -lasound
+	LDFLAGS += -L./raylib -lbrcmGLESv2 -lbrcmEGL
+	CFLAGS  += -I./raylib
 endif
 
 TARGET = $(EXENAME)$(EXE)
@@ -66,13 +75,13 @@ clean:
 	rm -f $(DELETEFILES)
 
 $(TARGET):  $(MUSAHIGENCFILES:%.c=%.o) $(.CFILES:%.c=%.o) a314/a314.o
-	$(CC) -o $@ $^ -O3 -pthread $(LFLAGS) -lm -lstdc++
+	$(CXX) -o $@ $^ -pthread $(LDFLAGS) -lm
 
 buptest: buptest.c gpio/ps_protocol.c
-	$(CC) $^ -o $@ -I./ -march=armv8-a -mfloat-abi=hard -mfpu=neon-fp-armv8 -O0
+	$(CC) $^ -o $@ -I./ $(MIN_CFLAGS) -O0
 
 a314/a314.o: a314/a314.cc a314/a314.h
-	$(CXX) -MMD -MP -c -o a314/a314.o -O3 a314/a314.cc -march=armv8-a -mfloat-abi=hard -mfpu=neon-fp-armv8 -O3 -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -I. -I..
+	$(CXX) -MMD -MP -c -o a314/a314.o a314/a314.cc $(MIN_CFLAGS) -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -I. -I..
 
 $(MUSASHIGENCFILES) $(MUSASHIGENHFILES): $(MUSASHIGENERATOR)$(EXE)
 	$(EXEPATH)$(MUSASHIGENERATOR)$(EXE)
